### PR TITLE
V2.0 Upgrade

### PR DESCRIPTION
**Pull request by gis2all**
Here are some updates:
- Upgrade all projects to .NET 6 and remove .NET Framework projects.
- Replace Newtonsoft.Json library with System.Text.Json.
- Update the ArcGISProToArcGISEarth add-in to use the ArcGIS Pro SDK for .NET 3.x.

Related issues https://github.com/Esri/arcgisearth-automation-api/issues/30  https://github.com/Esri/arcgisearth-automation-api/issues/28 https://github.com/Esri/arcgisearth-automation-api/issues/29

* [ ] Code Review @yong7743 
* [x] Code Review @xuew8412

### CODE REVIEW

File: src/ArcGISProToArcGISEarth/ArcGISProToArcGISEarth.csproj [hunk 1]
```diff
@@ -1,160 +1,100 @@
...
+      <CopyLocal>False</CopyLocal>
       <Private>False</Private>
     </Reference>
     <Reference Include="ArcGIS.Desktop.Extensions">
       <HintPath>C:\Program Files\ArcGIS\Pro\bin\Extensions\DesktopExtensions\ArcGIS.Desktop.Extensions.dll</HintPath>
+      <CopyLocal>False</CopyLocal>
+      <Private>False</Private>
+    </Reference>
+    <Reference Include="ArcGIS.Desktop.GeoProcessing">
+      <HintPath>C:\Program Files\ArcGIS\Pro\bin\Extensions\GeoProcessing\ArcGIS.Desktop.GeoProcessing.dll</HintPath>
```
Comment (yong7743) [inline src/ArcGISProToArcGISEarth/ArcGISProToArcGISEarth.csproj:64]: Does the addin need the GeoProcessing.dll?

File: src/ArcGISProToArcGISEarth/ArcGISProToArcGISEarth.csproj [hunk 2]
```diff
@@ -1,160 +1,100 @@
...
-       Other similar extension points exist, see Microsoft.Common.targets.
-  <Target Name="BeforeBuild">
-  </Target>
-  <Target Name="AfterBuild">
-  </Target>
-  -->
-  <Target Name="BeforeBuild">
-    <Error Text="C:\Program Files\ArcGIS\Pro\bin\Esri.ProApp.SDK.Desktop.targets not found." Condition="!Exists('C:\Program Files\ArcGIS\Pro\bin\Esri.ProApp.SDK.Desktop.targets')" />
-  </Target>
+  <Import Project="C:\Program Files\ArcGIS\Pro\bin\Esri.ProApp.SDK.Desktop.targets" Condition="Exists('C:\Program Files\ArcGIS\Pro\bin\Esri.ProApp.SDK.Desktop.targets') AND !Exists('Esri.ArcGISPro.Extensions.targets')" />
```
Comment (yong7743) [inline src/ArcGISProToArcGISEarth/ArcGISProToArcGISEarth.csproj:99]: I'm thinking if we can get the installation directory from the registry. e.g. `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\ESRI\ArcGISPro`

File: src/ArcGISProToArcGISEarth/HelpButton.cs [hunk 3]
```diff
@@ -12,16 +12,28 @@
...
         protected override void OnClick()
         {
-            // Open the help page. 
-            Process.Start("http://doc.arcgis.com/en/arcgis-earth/automation-api/samples.htm");
+            try
+            {
+                // Open the help page. 
+                Process.Start(new ProcessStartInfo
+                {
+                    FileName = "http://doc.arcgis.com/en/arcgis-earth/automation-api/samples.htm",
```
Comment (yong7743) [inline src/ArcGISProToArcGISEarth/HelpButton.cs:29]: Replace the HTTP address with the HTTPS:
https://doc.arcgis.com/en/arcgis-earth/automation-api/samples.htm

**Review by xuew8412 — APPROVED** [review]
LGTM

File: src/ArcGISProToArcGISEarth/ArcGISProToArcGISEarth.csproj [hunk 4]
```diff
@@ -7,6 +7,7 @@
     <AppendRuntimeIdentifierToOutputPath>false</AppendRuntimeIdentifierToOutputPath>
     <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
     <NoWarn>CA1416</NoWarn>
+    <ProInstallFolder>C:\Program Files\ArcGIS\Pro</ProInstallFolder>
```
Comment (yong7743) [inline src/ArcGISProToArcGISEarth/ArcGISProToArcGISEarth.csproj:10]: Rename `ProInstallFolder` to `ProInstallDir`

File: src/ArcGISProToArcGISEarth/README.md [hunk 5]
```diff
@@ -2,7 +2,7 @@
 Use ArcGIS Pro 3.0 or later versions to control and communicate with ArcGIS Earth.
 
 ### Build
-Build and compile the add-in using source codes provided. Modify codes where necessary in your own program.
+Build and compile the add-in using source codes provided. Modify codes where necessary in your own program. If your ArcGIS Pro installation directory is not `C:\Program Files\ArcGIS\Pro`, you need to modify the value of `ProInstallFolder` in the ArcGISProToArcGISEarth.csproj file to your custom installation directory.
```
Comment (yong7743) [inline src/ArcGISProToArcGISEarth/README.md:5]: ProInstallFolder -> ProInstallDir

**Review by yong7743 — APPROVED** [review]
looks good.

**Pull request merged**